BismarkUnchained/scenes/Player/Player.gd

```diff
@@ -10,21 +10,29 @@ var curr_velocity = Vector2(0, 0)
 
 # Bullet variables
 onready var bullet_sprite = preload("res://scenes/Weapons/Bullet/Bullet.tscn")
-
-const time_to_shoot = 0.75
 const bullet_velocity = 1000
 
 var time_last_shot = -1
 var last_dir = Vector2(0, 0)
 
+# Upgrades
+var upgrades = 0
+var maxhealth = 100
+var regen = 0
+var lifesteal = 0
+var time_to_shoot = 0.75
+
 # Player attribute variables
 var health = 100
 var experience = 0
 var level = 1
 
+# Regen variables
+var last_regen = 0
+
 func damage():
 	health -= 5
-	get_node('/root/Main/UI/Health').value = health
+	get_node('/root/Main/UI/Health').value = health * 100 / maxhealth
```

Comment:
we should probably have a wrapper for UI calls, so we don't have to go to specific UI elements - not an issue for now though